test/unit/tools/test_watcher.py

```diff
@@ -35,11 +36,12 @@ def test_tool_conf_watcher():
 
     callback = CallbackRecorder()
     conf_watcher = watcher.get_tool_conf_watcher(callback.call)
+    conf_watcher.start()  # Need to manually start conf_watcher thread
```

Comment:
This and the sleep should fix https://github.com/galaxyproject/galaxy/issues/2831